### PR TITLE
Add incremental load tests

### PR DESCRIPTION
**Pull request by katlapinka**

### CODE REVIEW

File: test/functional/tests/incremental_load/test_incremental_load.py [hunk 1]
```diff
@@ -0,0 +1,646 @@
...
+        plug_device = devices["core"]
+    with TestRun.step("Start cache and add core."):
+        cache = casadm.start_cache(cache_dev, cache_mode=CacheMode.WB, force=True)
+        core = cache.add_core(core_dev)
+    with TestRun.step("Create init config file using current CAS configuration."):
+        init_config.create_init_config_from_running_configuration()
+    with TestRun.step("Create filesystem on CAS device and mount it."):
+        core.create_filesystem(Filesystem.ext3)
+        core.mount(mount_dir)
+    with TestRun.step("Create a test file with random writes on mount point and count it's md5."):
```
Comment (imjfckm) [inline test/functional/tests/incremental_load/test_incremental_load.py]: I think you should set `seq_cutoff_policy` to `never` if you're using `dd` - most of the workload doesn't get cached

File: test/functional/tests/incremental_load/test_incremental_load.py [hunk 2]
```diff
@@ -0,0 +1,646 @@
...
+def test_attach_core_to_inactive_cache_volume():
+    """
+        title: Test for attaching device to inactive cache volume.
+        description: |
+          Try to attach core device to inactive cache volume and check if it is visible in OS
+          properly.
+        pass_criteria:
+          - No kernel error
+          - Core status change successfully
+          - Cache loads with inactive core device
```
[the remark this answers is not in the document]
Comment (katlapinka) [reply test/functional/tests/incremental_load/test_incremental_load.py]: Done

File: test/functional/tests/incremental_load/test_incremental_load.py [hunk 3]
```diff
@@ -0,0 +1,646 @@
...
+    with TestRun.step("Load cache."):
+        casadm.load_cache(cache_dev)
+    with TestRun.step("Check if list cache command shows proper output (cache should have status "
+                      "Incomplete and cache volume should be Inactive)."):
+        if core.get_status() != CoreStatus.inactive:
+            TestRun.fail("Core should be in inactive state.")
+        if cache.get_status() != CacheStatus.incomplete:
+            TestRun.fail("Cache should be in incomplete state.")
+    with TestRun.step("Plug missing device and stop cache."):
+        plug_device.plug()
```
[the remark this answers is not in the document]
Comment (katlapinka) [reply test/functional/tests/incremental_load/test_incremental_load.py]: Done

File: test/functional/tests/incremental_load/test_incremental_load.py (hunk 1 above)
Comment (katlapinka) [reply test/functional/tests/incremental_load/test_incremental_load.py]: Done

File: test/functional/tests/incremental_load/test_incremental_load.py [hunk 4]
```diff
@@ -0,0 +1,664 @@
...
+    with TestRun.step("Create init config file using current CAS configuration."):
+        init_config.create_init_config_from_running_configuration()
+    with TestRun.step("Stop cache."):
+        cache.stop()
+    with TestRun.step("Load cache."):
+        casadm.load_cache(cache_dev)
+    with TestRun.step("Check if there is CAS device in /dev and core is in active status."):
+        core.check_if_is_present_in_os()
+        if core.get_status() != CoreStatus.active:
+            TestRun.fail("Core should be in active state.")
```
Comment (micrakow) [inline test/functional/tests/incremental_load/test_incremental_load.py]: general - I think that current state could be logged as well

File: test/functional/tests/incremental_load/test_incremental_load.py [hunk 5]
```diff
@@ -0,0 +1,664 @@
...
+@pytest.mark.require_disk("core2", DiskTypeLowerThan("cache"))
+def test_flush_inactive_devices():
+    """
+        title: Test for flushing inactive CAS devices.
+        description: Validate that CAS prevents flushing dirty data from inactive CAS devices.
+        pass_criteria:
+          - No kernel error
+          - |
+            Flushing inactive CAS devices is possible neither by cleaning thread,
+            nor by calling cleaning methods
```
Comment (micrakow) [inline test/functional/tests/incremental_load/test_incremental_load.py]: '|' not needed if multiline is aligned.

File: test/functional/tests/incremental_load/test_incremental_load.py [hunk 6]
```diff
@@ -0,0 +1,664 @@
...
+        cache.stop()
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_list_cache_and_cache_volumes():
+    """
+        title: List cache with cache volumes and check their status.
+        description: |
+          Check if casadm command correctly lists caches and cache volumes with their status.
```
Comment (micrakow) [inline test/functional/tests/incremental_load/test_incremental_load.py]: statuses

File: test/functional/tests/incremental_load/test_incremental_load.py [hunk 7]
```diff
@@ -0,0 +1,664 @@
...
+    with TestRun.step("Stop cache."):
+        cache.stop()
+    with TestRun.step("Unplug core device."):
+        plug_device.unplug()
+    with TestRun.step("Load cache."):
+        output = TestRun.executor.run(cli.load_cmd(cache_dev.system_path))
+        cli_messages.check_msg(output, cli_messages.load_inactive_core_missing)
+    with TestRun.step("Plug missing device and stop cache."):
+        plug_device.plug()
+        core.wait_for_status_change(CoreStatus.active)
```
Comment (micrakow) [inline test/functional/tests/incremental_load/test_incremental_load.py]: hm?

File: test/functional/tests/incremental_load/test_incremental_load.py [hunk 8]
```diff
@@ -0,0 +1,664 @@
...
+    """
+        title: Print statistics for cache with inactive cache volumes.
+        description: |
+          Check if statistics are displayed properly when there is one or more
+          inactive cache volumes added to cache.
+        pass_criteria:
+          - No kernel error
+          - |
+            All statistics should contain appropriate information depend on situation of
+            cache and core devices (as described in test steps)
```
Comment (micrakow) [inline test/functional/tests/incremental_load/test_incremental_load.py]: as above

File: test/functional/tests/incremental_load/test_incremental_load.py (hunk 4 above)
Comment (katlapinka) [reply test/functional/tests/incremental_load/test_incremental_load.py]: Done

File: test/functional/tests/incremental_load/test_incremental_load.py (hunk 5 above)
Comment (katlapinka) [reply test/functional/tests/incremental_load/test_incremental_load.py]: Done

File: test/functional/tests/incremental_load/test_incremental_load.py (hunk 6 above)
Comment (katlapinka) [reply test/functional/tests/incremental_load/test_incremental_load.py]: Done

File: test/functional/tests/incremental_load/test_incremental_load.py (hunk 8 above)
Comment (katlapinka) [reply test/functional/tests/incremental_load/test_incremental_load.py]: Done

File: test/functional/tests/incremental_load/test_incremental_load.py [hunk 9]
```diff
@@ -0,0 +1,687 @@
...
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_load_cache_with_inactive_core():
+    """
+        title: Load cache with unavailable core devices.
+        description: Check if it is possible to load cache with unavailable core devices.
+        pass_criteria:
+          - No kernel error
+          - It is possible to perform cache load operation with unavailable devices.
+          - Warning message about not available core device should appear.
```
Comment (Deixx) [inline test/functional/tests/incremental_load/test_incremental_load.py]: - statuses

File: test/functional/tests/incremental_load/test_incremental_load.py [hunk 10]
```diff
@@ -0,0 +1,687 @@
...
+@pytest.mark.require_disk("core2", DiskTypeSet([DiskType.sata, DiskType.hdd, DiskType.hdd4k]))
+def test_print_statistics_inactive(cache_mode):
+    """
+        title: Print statistics for cache with inactive cache volumes.
+        description: |
+          Check if statistics are displayed properly when there is one or more
+          inactive cache volumes added to cache.
+        pass_criteria:
+          - No kernel error
+          - All statistics should contain appropriate information depend on situation of
```
Comment (Deixx) [inline test/functional/tests/incremental_load/test_incremental_load.py]: depending

**Pull request merged**